### DISCUSSION TITLE
Allow the use of negative values with prefetchExpires option

### DISCUSSION THREAD

**Discussion by alpominth** [0xERR0R/blocky#1005] [Ideas]
I had an idea for Blocky: Ability to select a negative value in **prefetchExpires** option.

The use case is to keep the cache updated without having to wait the expiration of a cached DNS query.

Let's suppose I configured Blocky to cache domains for 172800 seconds (2 days); if choosing -86400s (1 day) in **prefetchExpires**, the prefetch tracking windows would update DNS queries that  have received hits in the last 24 hours, or also if the cached domain has a TTL lower than the addition of [TTL] + (-prefetchExpires).

I think this is a good idea.

**Comment by alpominth** [comment]
@kwitsch @ThinkChaos @0xERR0R 

Did you find my suggestion reasonable to implement it?

At least for me, it would be useful to prefetch already queried entries in the cache with a negative value in **prefetchExpires**.

**Reply by kwitsch** [reply]
I'm currently not really familiar with the internals of of prefetching.
I'll take your suggestion into account next time working on it(on schedule for the redis rework).

**Comment by alpominth** [comment]
Another good reason to implement the feature I suggested: It does not require storing the cached domains in memory if using Redis.

When the DNS lookup is done, just find in the cache the entry of the domain, check if its TTL is lower than the addition of [TTL] + (-prefetchExpires), if it is, do the prefetch.

Very simple.
